### PR TITLE
Added Cross-Platform Compatibility

### PR DESCRIPTION
**Pull request by Dannygop**
Changes poc.py:
- Added: a bit more error handling
- Added: cross platform compatibility for shell
- Cleaned up code to be a bit more consistent

### CODE REVIEW

**Comment by ebadfd** [comment]
hi please resolve conflicts

**Comment by Dannygop** [comment]
Should be fixed now

**Comment by kozmer** [comment]
I don't currently have anywhere to test the Windows compatibility, is it okay if you make a PoC video w/ Windows and show me? Thank you :)

**Comment by svmorris** [comment]
Hi,
Yea sorry for getting back a bit late on this. I have 2 problems which are really slowing down this review process:
1: You have changed the file in many places, most of them not relevant to the purpose of the patch. ie things like this:
Screenshot from 2021-12-16 16-03-27

2. This is more of a fault of my own, but I don't have a windows computer with me, and we are waiting on @kozmer  to be free so he can test it.

File: poc.py [hunk 1]
```diff
@@ -32,107 +40,115 @@ def payload(userip , webport , lport):
...
+    # writing the exploit to Exploit.java file
+    try:
+        with open("Exploit.java", "w+") as exploit_file:
+            exploit_file.write(str(gen_exploit))
+        print(Fore.GREEN + '[+] Exploit java class created success')
+    except Exception as e:
+        print(Fore.RED + f'[-] Something went wrong {e.__str__()}')
+    check_java_available()
+    print(Fore.GREEN + '[+] Setting up the LDAP and web server.')
+    print(Fore.GREEN + f"[+] Starting the Web server on port {webport} http://0.0.0.0:{webport}", end="\n\n")
```
Comment (LennardTFD) [inline poc.py:97]: Due to change of parameter names (even though they are pretty inconsistent):

https://github.dev/Dannygop/log4j-shell-poc/blob/7f5d0bd7400f64cc4a4e1abc89e2d73eb8113a89/poc.py#L97 `webport` needs to be changed to `web_port` twice.

**Review by LennardTFD — CHANGES_REQUESTED** [review]

File: poc.py (hunk 1 above)
Comment (Dannygop) [reply poc.py:97]: ahh yes, completely overlooked that because I was just changing it on github and not in my ide.

Thank you for noticing

**Review by kozmer — COMMENTED** [review]
All works fine on my end, thanks :)

**Comment by Dannygop** [comment]
I'm sorry, the reason for all those small changes are, because I use PyCharm as IDE and it gives: spelling checks, syntax corrections and indentation corrections.
I pressed reformat file to apply those changes.

**Comment by ebadfd** [comment]
its all good we already fixed it :smiley: I think the java class you added gives a IllegalThreadStateException but the exploit still works on windows and unix 

```
java.lang.IllegalThreadStateException: process hasn't exited
        at java.lang.UNIXProcess.exitValue(UNIXProcess.java:289)
        at Exploit.<init>(Exploit.java:31)
        at Exploit.main(Exploit.java:9)
```

**Comment by Dannygop** [comment]
> its all good we already fixed it 😃 I think the java class you added gives a IllegalThreadStateException but the exploit still works on windows and unix
> 
> ```
> java.lang.IllegalThreadStateException: process hasn't exited
>         at java.lang.UNIXProcess.exitValue(UNIXProcess.java:289)
>         at Exploit.<init>(Exploit.java:31)
>         at Exploit.main(Exploit.java:9)
> ```

Where does it gives this? I only added another function which checks the OS name and returns a string back (sorry for the late reply)

Btw, was it intentional to revert the code to the old state?

**Comment by kozmer** [comment]
> > its all good we already fixed it smiley I think the java class you added gives a IllegalThreadStateException but the exploit still works on windows and unix
> > ```
> > java.lang.IllegalThreadStateException: process hasn't exited
> >         at java.lang.UNIXProcess.exitValue(UNIXProcess.java:289)
> >         at Exploit.<init>(Exploit.java:31)
> >         at Exploit.main(Exploit.java:9)
> > ```
> 
> Where does it gives this? I only added another function which checks the OS name and returns a string back (sorry for the late reply)
> 
> Btw, was it intentional to revert the code to the old state?

Yeah, we reverted back a few commits. Seems like the new exploit was causing issues within the Docker container when running the exploit. It didn't necessarily break things but it was making things quite unstable on my end which I noticed. Feel free to make another PR but could you test it/make a PoC of it working on Windows if you do? Thanks!

**Pull request merged**